prediction_market_agent/agents/microchain_agent/nft_treasury_game/scripts/generate_report.py

```
@@ -61,36 +62,52 @@ def summarize_past_actions_from_agent(agent_memories: list[LongTermMemories]) ->
-def summarize_prompts_from_all_agents() -> str:
+def store_all_learnings_in_db(
+    final_summary: str, learnings_per_agent: dict[AgentIdentifier, str]
+) -> None:
+    table_handler = ReportNFTGameTableHandler()
+    for agent_id, learnings in learnings_per_agent.items():
+        report = build_report(agent_id=agent_id, learnings=learnings)
+        logger.info(f"Saving report from {agent_id}")
+        table_handler.save_report(report)
+
+    final_report = build_report(agent_id=None, learnings=final_summary)
+    logger.info(f"Saving final summary")
```

Ruff (0.8.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)